doc/Programming-guide/Your-first-Kaa-application/index.md

```diff
@@ -399,6 +415,117 @@ int main(void)
+
+@end
+
+@implementation ViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+
+    NSLog(@"Your first application started");
+
+    // Create a Kaa client and add a listener which displays the Kaa client status
+    // as soon as the Kaa client is started.
+    self.kaaClient = [KaaClientFactory clientWithStateDelegate:self];
+    
+    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
+    NSString *configurationPath = [[paths objectAtIndex:0] stringByAppendingPathComponent:@"savedconfig.cfg"];
+
+    // Persist configuration in a local storage to avoid downloading it each time the Kaa client is started.
+    [self.kaaClient setConfigurationStorage:[SimpleConfigurationStorage storageWithPath:configurationPath]];
+    
+    // Add a listener which displays the Kaa client configuration each time it is updated.
+    [self.kaaClient addConfigurationDelegate:self];
+    
+    [self.kaaClient setProfileContainer:self];
+    
+    // Start the Kaa client and connect it to the Kaa server.
+    [self.kaaClient start];
+}
+
+#pragma mark - Supporting methods
+
+- (void)createSamplingTimerWithInterval:(NSInteger)period {
+    if (self.samplingTimer) {
+        [self.samplingTimer invalidate];
+    }
+    self.samplingTimer = [NSTimer scheduledTimerWithTimeInterval:period target:self selector:@selector(generateTemperature) userInfo:nil repeats:YES];
+}
+
+- (void)generateTemperature {
+    NSInteger upperTemperatureLimit = 32;
+    NSInteger lowerTemperatureLimit = 25;
+    NSInteger temperature = arc4random()%(upperTemperatureLimit - lowerTemperatureLimit) + lowerTemperatureLimit;
+    NSLog(@"Sampled temperature: %ld", temperature);
+    [self.kaaClient addLogRecord:[[KAADataCollection alloc] initWithTemperature:(int32_t)temperature]];
+}
+
+#pragma mark - Delegate methods
+
+- (void)onStarted {
+    NSLog(@"Kaa client started");
+    [self displayConfiguration];
+}
+
+- (void)onStopped {
+    NSLog(@"Kaa client stopped");
+}
+
+- (void)onStartFailureWithException:(NSException *)exception {
+    NSLog(@"START FAILURE: %@ : %@", exception.name, exception.reason);
+}
+
+- (void)onPaused {
+    NSLog(@"Client paused");
+}
+
+- (void)onPauseFailureWithException:(NSException *)exception {
+    NSLog(@"PAUSE FAILURE: %@ : %@", exception.name, exception.reason);
+}
+
+- (void)onResume {
+    NSLog(@"Client resumed");
+}
+
+- (void)onResumeFailureWithException:(NSException *)exception {
+    NSLog(@"RESUME FAILURE: %@ : %@", exception.name, exception.reason);
+}
+
+- (void)onStopFailureWithException:(NSException *)exception {
+    NSLog(@"STOP FAILURE: %@ : %@", exception.name, exception.reason);
+}
+
+- (KAAEmptyData *)getProfile {
+    return [[KAAEmptyData alloc] init];
+}
+
+- (void)onConfigurationUpdate:(KAAConfiguration *)configuration {
+    NSLog(@"Configuration was updated");
+    [self displayConfiguration];
+}
+
+- (void)displayConfiguration {
```

Comment:
Please, move this method into **Supporting methods** pragma block, 'cause it doesn't belong to any of the delegate methods.